ip_usage.go

```
@@ -32,16 +32,15 @@ func validateIPUsage(r *http.Request) error {
 	}
 	now := time.Now()
 	timeBeforeMinRequestInterval := now.Add(-minRequestInterval)
-	var count int
 	ip, err := ipFromRequest(r)
 	if err != nil {
 		return err
 	}
-	dbResult := db.Model(&ipUse{}).Where(&ipUse{IP: ip}).Where("last_use BETWEEN ? AND ?", timeBeforeMinRequestInterval, now).Count(&count)
-	dbErrors := dbResult.GetErrors()
-	if httpserverutils.HasDBError(dbErrors) {
-		return httpserverutils.NewErrorFromDBErrors("Some errors were encountered when checking the last use of an IP:", dbResult.GetErrors())
-	}
+	count, err := db.Model(&ipUse{}).
```

I think you are ignoring err here.